src/spell/modules/spell/spell/src/mailbox.rs

```diff
@@ -0,0 +1,243 @@
+use eyre::WrapErr;
+use marine_rs_sdk::marine;
+use marine_sqlite_connector::State;
+
+use fluence_spell_dtos::value::{StringValue, UnitValue};
+
+use crate::auth::{is_by_creator, is_by_spell};
+use crate::kv::primitive::read_string;
+use crate::schema::db;
+
+
+#[marine]
+pub struct AllMailboxResult {
+    pub messages: Vec<String>,
+    pub success: bool,
+    pub error: String,
+}
+
+#[marine]
+pub fn store_mailbox(message: String) -> UnitValue {
+    let call_parameters = marine_rs_sdk::get_call_parameters();
+
+    // We want to prevent anyone except this spell to store logs to its kv
+    if !is_by_creator() || !is_by_spell(&call_parameters) {
```

Comment:
isn't consistent with the comment. should be
```suggestion
    if !is_by_spell(&call_parameters) {
```

Reply:
yeah, but we should check that this particle id has been created by the spell too. Maybe I need to refactor these methods a bit